lib/pure/json.nim

```diff
@@ -1072,7 +1072,10 @@ when defined(nimFixedForwardGeneric):
 
   proc initFromJson[T](dst: var Option[T]; jsonNode: JsonNode; jsonPath: var string) =
     if jsonNode != nil and jsonNode.kind != JNull:
-      dst = some(default(T))
```

Comment:
It is a backport PR to 1.2 branch which is already on devel. The original PR => https://github.com/nim-lang/Nim/pull/15817

Reply:
> Thus we should fix Option[T] for version 2 instead of working around it.

I tested https://github.com/nim-lang/Nim/pull/18401 With that PR, the workaround can be removed.

Reply:
So I think 1-2 branch can live with this workaround and the devel branch can evolve with https://github.com/nim-lang/Nim/pull/18401 or similar PRs.

Reply:
Ok, agreed.

Reply:
Is there anything preventing this commit from being backported? Seems like the discussion about `Option` is not a blocker for that.

Reply:
I guess no if it could pass the CI.

Reply:
@xflywind I did a CI run with no changes(only a new line) and CI is failing again.

Reply:
It seems to be a pre-existing failure => https://github.com/nim-lang/Nim/tree/version-1-2

Reply:
> Hello @narimiran do you have any idea why [version-1-2](https://github.com/nim-lang/Nim/tree/version-1-2) branch failed?

It seems because some packages in the mean time have stopped supporting v1.2.x of Nim.

I'll disable them and then re-run the CIs for this PR.

Reply:
> I'll disable them and then re-run the CIs for this PR.

Done.

@yyoncho You can now revert the second commit.